format/src/main/thrift/schema.thrift

```diff
@@ -121,6 +127,24 @@ struct SchemaEvolution{
 }
 
 /**
+ * One partition in table
+ */
+struct Partition{
+    1: optional i32 partition_id;
+    2: required list<string> boundary_value_list;
+}
+
+/**
+ * Partition information of table
+ */
+struct PartitionInfo{
```

Comment:
To make this metadata more clear, can we do like this:
`struct PartitionInfo {
1: required PartitionType type;
2: optional RangePartitionInfo rangePartitionInfo;
3: optional ListPartitionInfo listPartitionInfo;
...
}
`
So that you can access each partition info based on the first member (type)